File: CodeJam/CJ_2023/Round_A/Untie.cpp
#include <iostream>
#include <cstring>
using namespace std;

const int MAXN = 1005;

int main() {
    int T;
    cin >> T;
    for (int t = 1; t <= T; t++) {
        char s[MAXN];
        cin >> s;
        int n = strlen(s);
        int ans = n;
        for (char c = 'R'; c <= 'S'; c++) {
            int cnt = 0;
            for (int i = 0; i < n; i++) {
                if (s[i] == c) {
                    if (s[(i+1)%n] == c || s[(i+n-1)%n] == c) cnt++;
                }
            }
            ans = min(ans, cnt);
        }
        cout << "Case #" << t << ": " << ans << endl;
    }
    return 0;
}

File: CodeJam/CJ_2023/Round_A/Untie.py
T = int(input())

for t in range(1, T+1):
    s = input()
    n = len(s)
    ans = n
    for c in "RPS":
        cnt = 0
        for i in range(n):
            if s[i] == c:
                if s[(i+1)%n] == c or s[(i+n-1)%n] == c:
                    cnt += 1
        ans = min(ans, cnt)
    print(f"Case #{t}: {ans}")
